inspirehep/base/templates/header.html

```diff
@@ -43,245 +42,172 @@
 
 <section>
   <nav id="topnav" class="navbar navbar-fixed-top navbar-inverse navbar-custom" role="navigation">
-    <div class="container">
-      <div class="navbar-header">
-        <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-ex1-collapse">
-          <span class="sr-only">{{ _("Toggle navigation") }}</span>
-          <span class="icon-bar"></span>
-          <span class="icon-bar"></span>
-          <span class="icon-bar"></span>
-        </button>
-        <a class="navbar-brand" href="{{ url_for('collections.index') }}">
-          <span id="{{ collection_name | sanitize_collection_name }}-collection">
-            {% include 'labs-logo.svg' %}
-          </span>
-        </a>
+    <div class="navbar-header">
+      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target=".navbar-ex1-collapse">
+        <span class="sr-only">{{ _("Toggle navigation") }}</span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <a class="navbar-brand" href="{{ url_for('collections.index') }}">
+        <span id="{{ collection_name | sanitize_collection_name }}-collection">
+          {% include 'labs-logo.svg' %}
+        </span>
+      </a>
+      {% if request.endpoint in ['search.search', 'record.metadata'] %}
+        <div id="top-search-bar-small" class="col-sm-5 {{collection.name}}">
+          {% include "search/form/form.html" %}
+        </div>
+      {% endif %}
+    </div>
+    {% if request.endpoint in ['search.search', 'record.metadata'] %}
+      <div id="top-search-bar-full" class="col-md-4 col-lg-5 {{collection.name}}">
+        {% include "search/form/form.html" %}
```

Comment:
The `top-search-bar-small` is included in the `navbar-header` div, whereas the `top-search-bar-full` is outside it, I found this necessary to accommodate for the mobile view. Open to suggestions for improvement though!